Automattic/wp-calypso
# Add-Ons: Retrieve site features from data-stores package

Fixes https://github.com/Automattic/wp-calypso/issues/87527

## Proposed Changes

Part of refactoring the add-ons metadata to migrate and be derived from data-stores. Part of uncoupling the plans-grid-next package from remaining Calypso dependencies.

* Introduces a `useSiteFeatures` hook under data-stores for retrieving site features from `/sites/:site/features` endpoint
* Uses that in place of local state
* TODO: Once features are migrated to `@automattic/calypso-products` we may have a `FeatureSlug` definition that we can incorporate (left some code TODOs for this on the `SiteFeatures` type)

## Testing Instructions



* Go to `/add-ons/[ site ]` and confirm the Jetpack AI add-on is not shown if the site already supports the feature
* Go to `/add-ons/[ paid site ]` and confirm `Included in your plan` is shown when plan already supports one of the add-ons

## Pre-merge Checklist
